Spawn terraform platform in CI for example solution

alexandre-allard opened the pull request:
**Component**: ci

**Context**: see #1856

**Summary**:

**Acceptance criteria**: 

---

Closes: #1856

alexandre-allard committed:
ci: add a new terraform worker for solutions testing

this worker is a single node bootstrap + a sidecar
VM to run the Cypress tests as they can really
consume host resources.

Refs: #1856

alexandre-allard committed:
ci: add a stage to test solutions CLI & UI

add a new stage single-node-solutions that
will tests both Solutions CLI and UI on
a CentOS 7 single node cluster, with a sidecar
VM that'll run the cypress tests to avoid
consuming the resources of the bootstrap node.

Refs: #1856

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/1856-spawn-tf-pf-example-solution` into
`development/2.4`
* [w/2.5/feature/1856-spawn-tf-pf-example-solution](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/feature/1856-spawn-tf-pf-example-solution) will be merged into `development/2.5` 
* [w/2.6/feature/1856-spawn-tf-pf-example-solution](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/feature/1856-spawn-tf-pf-example-solution) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

bert-e commented:
# History mismatch



Merge commit #4522403a52bf3cf801e15648a933cd9c21142e74 on the integration branch
`w/2.5/feature/1856-spawn-tf-pf-example-solution` is merging a branch which is neither the current
branch `feature/1856-spawn-tf-pf-example-solution` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `feature/1856-spawn-tf-pf-example-solution` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/1856-spawn-tf-pf-example-solution` into
`development/2.4`
* [w/2.5/feature/1856-spawn-tf-pf-example-solution](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/feature/1856-spawn-tf-pf-example-solution) will be merged into `development/2.5` 
* [w/2.6/feature/1856-spawn-tf-pf-example-solution](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/feature/1856-spawn-tf-pf-example-solution) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

bert-e commented:
# History mismatch



Merge commit #5db0e3456f8ab99cb4f81ed572598109bfa09fb2 on the integration branch
`w/2.5/feature/1856-spawn-tf-pf-example-solution` is merging a branch which is neither the current
branch `feature/1856-spawn-tf-pf-example-solution` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `feature/1856-spawn-tf-pf-example-solution` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

Ebaneck commented:
/wait

alexandre-allard committed:
tests: retry pillar get in Solutions tests

Since we now need to restart the salt-master after
importing a new Solution, because we mount the
archive content inside the salt-master container
to make its CRDs available, we have to add a retry
mechanism on this salt command, otherwise it can
fail if the master is not yet ready.

Refs: #1856

alexandre-allard committed:
ci: add testrail install section

this section contains only single node solutions
installation on CentOS7 for now

Refs: #1856

alexandre-allard commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
